app/client/components/delivery/records/deliveryRecordsTable.tsx

```diff
@@ -0,0 +1,251 @@
+import { css } from "@emotion/core";
+import { palette, space } from "@guardian/src-foundations";
+import { textSans } from "@guardian/src-foundations/typography";
+import moment from "moment";
+import React, { useState } from "react";
+import { DeliveryRecordsDetail } from "../../../../shared/productResponse";
+import { DeliveryRecordsApiItem } from "../../../../shared/productResponse";
+import { minWidth } from "../../../styles/breakpoints";
+import { DeliveryRecordMessage } from "./deliveryRecordMessage";
+import { RecordAddress } from "./deliveryRecordsAddress";
+import { DeliveryProblemMap } from "./deliveryRecordsApi";
+import { PaginationNav } from "./deliveryRecordsPaginationNav";
+import { RecordStatus } from "./deliveryRecordStatus";
+
+interface RecordsTableProps {
+  data: DeliveryRecordsDetail[];
+  deliveryProblemMap: DeliveryProblemMap;
+  resultsPerPage: number;
+}
+
+export const RecordsTable = (props: RecordsTableProps) => {
+  const totalPages = Math.ceil(props.data.length / props.resultsPerPage);
+  const [currentPage, setCurrentPage] = useState(0);
+  const scrollToTop = () => {
+    window.scrollTo(0, 0);
+  };
+  const isOdd = (n: number) => Math.abs(n % 2) === 1;
```

Comment:
could this be done with css https://www.w3.org/Style/Examples/007/evenodd.en.html (probably with `nth-of-type` though to avoid react console warnings)

Reply:
The isOdd rule that I have used here is slightly more involved than the evenodd or nth-type css selectors as in some circumstances we have grouped tr elements (Where there is a delivery record problem or alert message). In this scenario instead of the following:

`even
odd
even
odd`

the correct rule might be:

`even
odd
odd
even
odd`

Reply:
ahh make sense!